### DISCUSSION TITLE
I made a `chezmoi`-like module that is more suited for public use images

### DISCUSSION THREAD

**Discussion by noahdotpy** [blue-build/.github#32] [Show and tell]
This module, instead of using external repositories, uses something like `config/homefiles/common` in your bluebuild repo.

`config/homefiles/common` in this case, would be a chezmoi source (initialised by `chezmoi init`, for example).

## This is how it works:

Step 1.
  - Copy systemd service, `.path` file, and script used by the service. This service is a user service and just does a `chezmoi apply` (with some arguments)
Step 2.
  - If `disable-service: true` then disable the service by default, otherwise enable
Step 3.
  - Copy all entries listed into (`add`: see README.md in modules/homefiles for example) to /usr/share/bluebuild/homefiles/$entry
Step 4.
  - Applies this entry to destination at `/etc/skel`

## Systemd Service - How does it work?

- This service runs at login, and whenever `/usr/share/bluebuild/homefiles/` changes

### What does the script do?

- The service runs `chezmoi apply` for every entry.
- I use `yes "skip" | chezmoi apply ...` to automatically skip any files that are different from the source.
- state file is set to `~/.local/state/bluebuild/homefiles/($entry_name)/chezmoi-state`
- `--no-tty --keep-going` is also tacked on, idk what it does but someone else used it so I did.

[noahdotpy/myublue: modules/homefiles](https://github.com/noahdotpy/myublue/tree/main/modules/homefiles)

**Comment by fiftydinar** [comment]
Thanks for sharing it with us!

Maybe some of this can be incorporated into already existing `chezmoi` module.

Like an option between choosing an external repository or internal location in `/config/chezmoi/`

**Comment by xynydev** [comment]
What's the reason for using chezmoi if no repository is used?

**Reply by noahdotpy** [reply]
> What's the reason for using chezmoi if no repository is used?

It's existing technology already made for this purpose.

Chezmoi is literally made for this exact purpose, home config file management. Chezmoi has useful features like templating, which is especially useful if there is no way you could get the user's username because it is just a plain string variable in a config file. Chezmoi also has the ability to make files executable with a prefix in the file name. AFAIK this would be pretty difficult to implement, probably requiring extra config for each file.

**Comment by noahdotpy** [comment]
btw im loving using nushell for all my little (or big) scripts
